scripts/tests/linux/log_line_processing.py

```diff
@@ -63,26 +63,28 @@ def _io_thread(self):
         err_wait = select.poll()
         err_wait.register(self.process.stderr, select.POLLIN | select.POLLHUP)
 
-        with open(self.output_path, "wt") as f:
+        with open(self.output_path, "wt", buffering=1) as f:  # Enable line buffering for immediate output
             f.write("PROCESS START: %s\n" % time.ctime())
+            f.flush()
             while not self.done:
-                changes = out_wait.poll(0.1)
-                if changes:
-                    out_line = self.process.stdout.readline()
-                    if not out_line:
-                        # stdout closed (otherwise readline should have at least \n)
-                        continue
-                    f.write(out_line)
-                    self.output_lines.put(out_line)
-
-                changes = err_wait.poll(0)
-                if changes:
+                out_changes = out_wait.poll(0.1)
```

Comment:
since we do the `filno in ...` thing, do we need an out and err wait that are separate anymore? it seems we could have a single poll object and maybe even poll for longer (the only reason we timeout the poll seems to stop ourselves)

Reply:
Discussed implementation changes offline. Marking this as resolved since we will move forward with removing the polling logic. Instead we will now create two threads (stdout and stderr) and read all lines.